silo-vaults/contracts/MetaMorpho.sol

```diff
@@ -480,6 +482,23 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         emit EventsLib.Skim(_msgSender(), token, amount);
     }
 
+    function claimRewards() public {
+        // TODO we can use supplyQueue as markets inputs, but what if we change queue and we still have rewards?
+        // I think it is better to let INCENTIVES_MODULE decide about markets
+        // (address[] memory logics) = INCENTIVES_MODULE.getMarketsIncentivesClaimingLogics(supplyQueue);
```

Comment:
> but what if we change queue and we still have rewards?

We can claim rewards when queue updated too.
I would go with `getMarketsIncentivesClaimingLogics(supplyQueue)`.

For example it is possible that some market will be removed from the vault but not from the incentives module. We will be wasting a gas trying to claim rewards for it.

Reply:
it works both ways, but if we ask question "where is the place for claiming logic?" it is not in vault, vault manages should not think about claiming logic, it should manage vault deposits. Also if we will need to add market only for claiming to the withdraw queue it is agains logic imo. os I would vote to menage markets in `INCENTIVES_MODULE`

Reply:
Both ways work for me.